packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -2105,17 +2105,10 @@ export class Bucket extends BucketBase {
       return undefined;
     }
 
-    if (
-      // KMS can't be used for logging since the logging service can't use the key - logs don't write
-      // KMS_MANAGED can't be used for logging since the account can't access the logging service key - account can't read logs
-      (!props.serverAccessLogsBucket && (
-        props.encryptionKey ||
-        props.encryption === BucketEncryption.KMS_MANAGED ||
-        props.encryption === BucketEncryption.KMS )) ||
-      // Another bucket is being used that is configured for default SSE-KMS
-      props.serverAccessLogsBucket?.encryptionKey
-    ) {
-      throw new Error('SSE-S3 is the only supported default bucket encryption for Server Access Logging target buckets');
+    // Proper permission can be added to the key and bucket for BucketEncryption.KMS case.
+    // But for BucketEncryption.KMS_MANAGED, the key permission cannot be modified.
+    if (!props.serverAccessLogsBucket && props.encryption === BucketEncryption.KMS_MANAGED) {
+      throw new Error('Default bucket encryption with KMS managed key is not supported for Server Access Logging target buckets');
```

Comment:
If this is the case, we should have integ tests where `encryption` is set to `BucketEncryption.KMS` with an `encryptionKey` and without an `encryptionKey` set.

Reply:
The case you talked about is checked under function `parseEncryption`.
```
    // if encryption key is set, encryption must be set to KMS.
    if (encryptionType !== BucketEncryption.KMS && props.encryptionKey) {
      throw new Error(`encryptionKey is specified, so 'encryption' must be set to KMS (value: ${encryptionType})`);
    }
```

The change I modified here is mean to fail the build when the server access logs bucket being set as itself and the bucket encryption is set to KMS_MANAGED. In this case, though the resource can be successfully deployed, we cannot modified the encryption key owned by KMS. The key and the bucket need proper permission to make the whole logging workflow work. So, people won't able to get access log from the bucket or the log would be unreadable.

This change won't affect snapshot at all.